luxtronik/__init__.py

```diff
@@ -1,21 +1,44 @@
 """luxtonik heatpump interface."""
 # -*- coding: utf-8 -*-
+# region Imports
+from __future__ import annotations
 
 import logging
 import socket
 import struct
+import threading
+import time
 
 from luxtronik.calculations import Calculations
 from luxtronik.parameters import Parameters
 from luxtronik.visibilities import Visibilities
+# endregion Imports
 
 LOGGER = logging.getLogger("Luxtronik")
 
+def is_socket_closed(sock: socket.socket) -> bool:
```

Comment:
Looking at the description of this pull request and the included changes, I can see that something is not correct.

The descriptions is talking about new datatypes, while the changes are also changing/modifying/improving the socket handling.

From my point of view this should be separated into multiple pull requests. It seems to me that for the datatypes only changes in luxtronik/datatypes.py and luxtronik/parameters.py are needed.

The changes in luxtronik/__init__.py are most likely referring to #36 and should be handled on their own.

Reply:
Yes, sorry. The changes in __init__.py should not included in this pr. I have pushed it after the pr...
Can I revert this commit or delete it?

Reply:
Technically the correct thing to fix this is to rebase it. Basically you have to do something along those lines:

git clone git@github.com:Bouni/python-luxtronik.git
cd python-luxtronik
git remote add benpru git@github.com:BenPru/python-luxtronik.git
git checkout -b benpru/main
git rebase -i origin/main 

Here you could delete the commits that are not relevant.

However the main problem is/was that you've created a MR from the `main` branch of your repository. Typically you should try to avoid that and create merge requests from "feature branches" into `main`. When `main` is the source, you cannot easily add additional commits.